lib/temporal/connection/grpc.rb

```diff
@@ -420,16 +420,66 @@ def get_search_attributes
         raise NotImplementedError
       end
 
-      def respond_query_task_completed
-        raise NotImplementedError
+      def respond_query_task_completed(result:, **request_params)
+        request = Temporal::Api::WorkflowService::V1::RespondQueryTaskCompletedRequest.new(
+          **request_params.merge(query_result: to_query_payloads(result))
```

Comment:
Minor note but this usage of collecting keyword args doesn't match the style of the rest of this file which explicitly lists each arg. This may be a blocker if @antstorm feels strongly about keeping the style consistent if a bit more verbose.

Reply:
Yeah, good point! The main idea was no to leak GRPC-specific stuff beyond this file. Return values do cross this boundary, but I think we should at least keep that separation for input arguments